### PR TITLE
Fix SNAPSHOT srcs/docs with Ivy upgrade.

### LINKED ISSUE #1750: updateClassifiers doesn't download sources and docs for updated snapshot dependencies
Hello.
sbt.version=0.13.7
we are using corporate maven repository (served by artifactory 3.4.1)
Our libraryDependencies:

``` scala
libraryDependencies ++= {
    val jettyVer = "9.2.3.v20140905"
    val akkaVersion = "2.3.7"
    val kamonVersion = "0.3.5"
    Seq(
        "javax.servlet" % "javax.servlet-api" % "3.1.0" % "provided",
        "com.simplesys" %% "akka-extender" % "1.0.2-SNAPSHOT",
        "com.simplesys" %% "isc-wrapper" % "1.0.0-SNAPSHOT",
        "com.simplesys" %% "common" % "1.0.0-SNAPSHOT",
        "com.simplesys.jdbc.drivers" % "oracle" % "11.2.0.4",
        "com.simplesys" %% "isc-components" % "1.0.0-SNAPSHOT",
        "com.simplesys" %% "core-library" % "1.0.0-SNAPSHOT",
        "com.simplesys" %% "scala-io-extender" % "1.0.0-SNAPSHOT",
        "ru.simplesys" %% "core-domains" % "1.2.1-SNAPSHOT",
        "ru.simplesys" %% "core-utils" % "1.0.1-SNAPSHOT",
        "com.simplesys" % "smartclient-js" % "10.0-v20141114",
        "com.simplesys" %% "scala-gen" % "1.0.0-SNAPSHOT" % "test",
        "com.typesafe.akka" %% "akka-actor" % akkaVersion,
        "com.typesafe.akka" %% "akka-slf4j" % akkaVersion,
        "com.typesafe.akka" %% "akka-persistence-experimental" % akkaVersion exclude("org.iq80.leveldb","leveldb"),
        "org.iq80.leveldb" % "leveldb" % "0.7",
        "com.twitter" %% "chill-akka" % "0.5.0",
        "com.github.bseibel" %% "akka-persistence-bdb" % "1.0.1",
        "com.typesafe.akka" %% "akka-testkit" % akkaVersion % "test",
        "org.eclipse.jetty" % "jetty-webapp" % jettyVer % "container",
        "org.eclipse.jetty" % "jetty-annotations" % jettyVer % "container",
        "org.eclipse.jetty" % "jetty-plus" % jettyVer % "container",
        "org.scalatest" %% "scalatest" % "2.2.2" % "test",
        "org.scalaj" %% "scalaj-http" % "0.3.16" % "test",
        "org.json4s" %% "json4s-jackson" % "3.2.10" % "test",
        "org.scalaz" %% "scalaz-core" % "7.1.0",
        "com.simplesys.edu.mit" % "jwi" % "2.3.3",
        "org.parboiled" %% "parboiled" % "2.0.1",
        "io.kamon" %% "kamon-core" % kamonVersion,
        "io.kamon" %% "kamon-statsd" % kamonVersion,
        "io.kamon" %% "kamon-system-metrics" % kamonVersion)
```

during updateClassifiers only jars are downloaded, without sources and docs, although they all have published sources and docs. This is really annoying, I'm ready to supply all info you need.
We are using 

``` scala
addSbtPlugin("no.arktekk.sbt" % "aether-deploy" % "0.13")
```

for publishing

updateClassifiers log:

```
sbt (dm-processing)> updateClassifiers
[info] Resolving org.ow2.asm#asm-tree;5.0.1 ...
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/akka-extender_2.11/1.0.2-SNAPSHOT/akka-extender_2.11-1.0.2-20141125.141707-139.jar ...
[info]  [SUCCESSFUL ] com.simplesys#akka-extender_2.11;1.0.2-SNAPSHOT!akka-extender_2.11.jar (1006ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/servlet-wrapper_2.11/1.0.0-SNAPSHOT/servlet-wrapper_2.11-1.0.0-20141125.141655-84.jar ...
[info]  [SUCCESSFUL ] com.simplesys#servlet-wrapper_2.11;1.0.0-SNAPSHOT!servlet-wrapper_2.11.jar (1120ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/xml-extender_2.11/1.0.0-SNAPSHOT/xml-extender_2.11-1.0.0-20141125.141653-18.jar ...
[info]  [SUCCESSFUL ] com.simplesys#xml-extender_2.11;1.0.0-SNAPSHOT!xml-extender_2.11.jar (743ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/logback-wrapper_2.11/1.0.0-SNAPSHOT/logback-wrapper_2.11-1.0.0-20141125.141653-20.jar ...
[info]  [SUCCESSFUL ] com.simplesys#logback-wrapper_2.11;1.0.0-SNAPSHOT!logback-wrapper_2.11.jar (785ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/common_2.11/1.0.0-SNAPSHOT/common_2.11-1.0.0-20141125.141653-33.jar ...
[info]  [SUCCESSFUL ] com.simplesys#common_2.11;1.0.0-SNAPSHOT!common_2.11.jar (895ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/joda-wrapper_2.11/1.0.0-SNAPSHOT/joda-wrapper_2.11-1.0.0-20141125.141653-20.jar ...
[info]  [SUCCESSFUL ] com.simplesys#joda-wrapper_2.11;1.0.0-SNAPSHOT!joda-wrapper_2.11.jar (760ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/saxon-wrapper_2.11/1.0.0-SNAPSHOT/saxon-wrapper_2.11-1.0.0-20141125.141653-18.jar ...
[info]  [SUCCESSFUL ] com.simplesys#saxon-wrapper_2.11;1.0.0-SNAPSHOT!saxon-wrapper_2.11.jar (718ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/scala-io-extender_2.11/1.0.0-SNAPSHOT/scala-io-extender_2.11-1.0.0-20141125.141653-19.jar ...
[info]  [SUCCESSFUL ] com.simplesys#scala-io-extender_2.11;1.0.0-SNAPSHOT!scala-io-extender_2.11.jar (707ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/ru/simplesys/core-utils_2.11/1.0.1-SNAPSHOT/core-utils_2.11-1.0.1-20141125.140620-6.jar ...
[info]  [SUCCESSFUL ] ru.simplesys#core-utils_2.11;1.0.1-SNAPSHOT!core-utils_2.11.jar (663ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/json-extender-typesafe_2.11/1.0.0-SNAPSHOT/json-extender-typesafe_2.11-1.0.0-20141125.141655-28.jar ...
[info]  [SUCCESSFUL ] com.simplesys#json-extender-typesafe_2.11;1.0.0-SNAPSHOT!json-extender-typesafe_2.11.jar (872ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/isc-misc_2.11/1.0.0-SNAPSHOT/isc-misc_2.11-1.0.0-20141125.141653-33.jar ...
[info]  [SUCCESSFUL ] com.simplesys#isc-misc_2.11;1.0.0-SNAPSHOT!isc-misc_2.11.jar (863ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/bonecp-wrapper_2.11/1.0.0-SNAPSHOT/bonecp-wrapper_2.11-1.0.0-20141125.141653-18.jar ...
[info]  [SUCCESSFUL ] com.simplesys#bonecp-wrapper_2.11;1.0.0-SNAPSHOT!bonecp-wrapper_2.11.jar (954ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/config-wrapper_2.11/1.0.0-SNAPSHOT/config-wrapper_2.11-1.0.0-20141125.141653-21.jar ...
[info]  [SUCCESSFUL ] com.simplesys#config-wrapper_2.11;1.0.0-SNAPSHOT!config-wrapper_2.11.jar (704ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/isc-wrapper_2.11/1.0.0-SNAPSHOT/isc-wrapper_2.11-1.0.0-20141125.141706-38.jar ...
[info]  [SUCCESSFUL ] com.simplesys#isc-wrapper_2.11;1.0.0-SNAPSHOT!isc-wrapper_2.11.jar (3529ms)
[info] downloading http://toucan.simplesys.lan/artifactory/repo/com/simplesys/scala-gen_2.11/1.0.0-SNAPSHOT/scala-gen_2.11-1.0.0-20141125.141713-20.jar ...
[info]  [SUCCESSFUL ] com.simplesys#scala-gen_2.11;1.0.0-SNAPSHOT!scala-gen_2.11.jar (1068ms)
[success] Total time: 146 s, completed 26.11.2014 12:22:46
sbt (dm-processing)> 
```

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #1750.

---

Updates to the fixes in sbt/ivy#17.

I realise this is missing some due diligence regression suite tests and notes,
but I wanted to push this out for feedback.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by dwijnand** [comment]
Copied from sbt/ivy#15, this is how I've been reproducing this bug:
- Stock Nexus installed with homebrew
- `sbt-gh1750-bippy`, the SNAPSHOT dependency
- `sbt-gh1750-myapp`, the project trying to get updated SNAPSHOT sources
- Change `bippy`, and `publish` to Nexus
- Run `updateClassifiers` in `myapp`, see if it downloads the sources jar
- As a second test, check the contents `bippy`:

```
unzip -c ~/.ivy2/cache/com.dwijnand/sbt-gh1750-bippy_2.11/srcs/sbt-gh1750-bippy_2.11-0.1.0-SNAPSHOT-sources.jar Bippy.scala
```

``` scala
// sbt-gh1750-bippy/build.sbt

organization := "com.dwijnand"
        name := "sbt-gh1750-bippy"
     version := "0.1.0-SNAPSHOT"
scalaVersion := "2.11.7"

val localNexusReleases  = "local-nexus-releases"  at "http://localhost:8081/nexus/content/repositories/releases"
val localNexusSnapshots = "local-nexus-snapshots" at "http://localhost:8081/nexus/content/repositories/snapshots"
val localNexusCreds = Credentials("Sonatype Nexus Repository Manager", "localhost", "admin", "admin123")

publishTo := Some(if (isSnapshot.value) localNexusSnapshots else localNexusReleases)
credentials := Seq(localNexusCreds)
```

``` scala
// sbt-gh1750-myapp/build.sbt
organization := "com.dwijnand"
        name := "sbt-gh1750-myapp"
     version := "0.1.0-SNAPSHOT"
scalaVersion := "2.11.7"

val localNexusReleases  = "local-nexus-releases"  at "http://localhost:8081/nexus/content/repositories/releases"
val localNexusSnapshots = "local-nexus-snapshots" at "http://localhost:8081/nexus/content/repositories/snapshots"

resolvers += localNexusReleases
resolvers += localNexusSnapshots

libraryDependencies += "com.dwijnand" %% "sbt-gh1750-bippy" % "0.1.0-SNAPSHOT"
```

**Comment by dwijnand** [comment]
Travis failed because sbt/ivy#17 hasn't been released (or merged) yet.

**Comment by dwijnand** [comment]
Rebased and force pushed.

Added a scripted test, which prior to the Ivy update fails and after passes(!).

However, as mentioned on Gitter, I needed a way to declare a local filesystem resolver that actually copied files out to the Ivy cache, instead of just using the file in place, so I could verify against what was resolved/downloaded.

So I've made a change to `MavenRepository` to allow for that. I've simulated and tested in a separated repo with MiMa that these are the changes necessary for bincompat, but please have a look.

I'm not sure if that flag needs notes. Also potentially I could split that change out of this pull request.

Still haven't written up notes for the fix itself; what do you think of it?

File: ivy/src/main/scala/sbt/ConvertResolver.scala [hunk 1]
```diff
@@ -120,7 +120,7 @@ private[sbt] object ConvertResolver {
               }
             }
             val resolver = new PluginCapableResolver
-            resolver.setRepository(new LocalIfFileRepo)
+            if (repo.localIfFile) resolver.setRepository(new LocalIfFileRepo)
```
Comment (dwijnand) [inline ivy/src/main/scala/sbt/ConvertResolver.scala:123]: Not sure about the name. I went with the prior art of the repo here.

**Comment by eed3si9n** [comment]
Commented on dwijnand/sbt@24495be35db71d4ad87399001cb37a94b8e826a7

**Comment by dwijnand** [comment]
@eed3si9n I've added another commit as there was a failure in Travis that I'm not sure if it was legitimate or not.

What do you think of https://github.com/dwijnand/sbt/commit/467e9ab9c248e044de3ba8d6fa7c93bf3e4f7637? Note I added a comment below it as well.

Other than that, what do you think of this PR, the fix itself etc?

**Comment by dwijnand** [comment]
Hmm yeah... looks like my test doesn't work on Travis:

```
Failed to read jar /tmp/sbt_8f770495/update-classifiers-snapshot-srcs/target/ivy/cache/org.example/bippy_2.11/srcs/bippy_2.11-1.0-SNAPSHOT-sources.jar
```

Don't understand why though :-/

**Comment by dwijnand** [comment]
Travis is failing when using `mavenResolverPlugin`. I have a local change that fixes the pathing to sources jar, but running that test shows that the issue is still present when using that plugin.. :(

**Comment by dwijnand** [comment]
Travis passed! :)

**Comment by jsuereth** [comment]
LGTM

**Comment by dwijnand** [comment]
Uh... I was going to make the change discussed in https://github.com/sbt/ivy/pull/17 and the relevant sbt changes here to enable the ivySettings flag..

I'll open a new PR in both repos with that change.

**Comment by dwijnand** [comment]
Also this was missing notes, so that'll be part of the follow up PR.

**Pull request merged**